plugins/coverimage.koplugin/main.lua

```diff
@@ -0,0 +1,190 @@
+            -- menu entry: filename dialog
+            {
+                text_func = function()
+                    return self.cover_image_path and T(_("Cover Image (%1)"), self.cover_image_path)
+                        or _("Cover Image (none)")
+                end,
+                keep_menu_open = true,
+                callback = function(menu)
+                    local InputDialog = require("ui/widget/inputdialog")
+                    local sample_input
+                    sample_input = InputDialog:new{
+                        title = _("Filename for Cover-Image"),
+                        input = self.cover_image_path,
+                        input_type = "string",
+                        description = _("You can enter the filename of the cover image here."),
+                        buttons = {
+                            {
+                                {
+                                    text = _("Cancel"),
+                                    callback = function()
+                                        UIManager:close(sample_input)
+                                    end,
+                                },
+                                {
+                                    text = _("Save"),
+                                    is_enter_default = true,
+                                    callback = function()
+                                        local new_cover_image_path = sample_input:getInputText()
+                                        if new_cover_image_path ~= self.cover_image_path then
+                                            self:cleanUpImage() -- with old filename
+                                            self.cover_image_path = new_cover_image_path -- update filename
+                                            G_reader_settings:saveSetting("cover_image_path", self.cover_image_path)
+                                            self:onReaderReady(self.ui.doc_settings) -- with new filename
+                                        end
+                                        UIManager:close(sample_input)
+                                        menu:updateItems()
+                                    end,
+                                },
+                            }
+                        },
+                    }
+                    UIManager:show(sample_input)
+                    sample_input:onShowKeyboard()
+                end
+            },
+            -- menu entry: enable
+            {
+                text_func = function()
+                    return _("Save book cover")
+                end,
+                checked_func = function()
+                    return self:_enabled()
+                end,
+                callback = function()
+                    self.enabled = not self.enabled
+                    G_reader_settings:saveSetting("cover_image_enabled", self.enabled)
+                    if self.enabled then
+                        self.restore = true
```

Comment:
Can't one toggle "enabled" while still wanting "restore" to stay false?
I assume the combination of `enabled=true` and `restore=false` is when one wants to keep the last book cover as the screensaver when exiting KOReader, or exiting it while in Filebrowser ?